versions.go

```diff
@@ -73,43 +86,90 @@ type LegacyMongoDBVersion struct {
+	return false
+}
+
+// IsDevelopmentSeries is not applicable to new versions, so always return false.
+func (v *NewMongoDBVersion) IsDevelopmentSeries() bool {
+	return false
+}
+
+// IsInitialStableReleaseCandidate is not applicable to new versions, so always return false.
+func (v *NewMongoDBVersion) IsInitialStableReleaseCandidate() bool {
+	return false
+}
+
+// StableReleaseSeries is not applicable to new versions, so always return the empty string.
+func (v *NewMongoDBVersion) StableReleaseSeries() string {
+	return ""
+}
+
+// IsLTS returns true if this is the year
+func (v *NewMongoDBVersion) IsLTS() bool {
+	return v.IsRelease() && v.Parsed().Minor == 0
+}
+
+// func IsContinuousRelease returns true if the version is a continuous release.
+func (v *NewMongoDBVersion) IsContinuousRelease() bool {
+	return v.IsRelease() && v.Parsed().Minor != 0
+}
+
+// IsDevelopmentRelease returns true if the version is a development release.
+func (v *NewMongoDBVersion) IsDevelopmentRelease() bool {
+	return v.isDevRelease
+}
+
+// DevelopmentReleaseNumber returns the number of the development release,
+// or -1 if not applicable.
+func (v *NewMongoDBVersion) DevelopmentReleaseNumber() int {
+	return v.devReleaseNumber
 }
 
 // CreateMongoDBVersion returns an implementation of the MongoDBVersion.
 // If the parsed version is before 4.5.0, then we use the legacy structure.
 // Otherwise, we use the modern versioning scheme.
 func CreateMongoDBVersion(version string) (MongoDBVersion, error) {
 	endOfLegacyVersion, _ := semver.Parse(endOfLegacy)
-
-	// pre-processing to then determine which version to use
-	toParse := version
-	if strings.HasSuffix(version, "-") && !strings.Contains(version, "pre") {
-			toParse += "pre-"
-	}
-	if strings.Contains(toParse, "~") {
-		versionParts := strings.Split(version, "~")
-		toParse = versionParts[0]
-	}
-
-	parsed, err := semver.Parse(toParse)
+	v, err := createLegacyMongoDBVersion(version)
 	if err != nil {
-		return nil, errors.Wrapf(err, "error parsing %s'", toParse)
+		return nil, errors.Wrapf(err, "creating initial version")
 	}
-	if parsed.LT(endOfLegacyVersion) {
-		return createLegacyMongoDBVersion(version)
+	if v.Parsed().LTE(endOfLegacyVersion) {
+		return v, nil
 	}
-	return createNewMongoDBVersion(version)
+	return createNewMongoDBVersion(*v)
 }
 
-
 // createNewMongoDBVersion takes a string representing a MongoDBVersion and
 // returns a NewMongoDBVersion object. All parsing of a version happens during this phase.
-func createNewMongoDBVersion(version string) (*NewMongoDBVersion, error) {
-	return nil, errors.New("not yet implemented")
+func createNewMongoDBVersion(parsedVersion LegacyMongoDBVersion) (*NewMongoDBVersion, error) {
+	v := &NewMongoDBVersion{LegacyMongoDBVersion: parsedVersion, devReleaseNumber: -1}
+	var err error
+	if strings.Contains(v.tag, devReleaseTag) {
+		v.isDevRelease = true
+		v.quarter = v.String()[:3]
+		v.devReleaseNumber, err = strconv.Atoi(v.tag[len(devReleaseTag):])
```

Comment:
Can we return immediately if the error is non-nil?

Reply:
We do return on the next line so I think it's fine; this is also the process with the existing LegacyCode

Reply:
I prefer to return the zero value if I return an error because usually the object is in an unusable intermediate state, but don't feel strongly about it.